### ISSUE TITLE
Ternary Operator not always parsable

### ISSUE THREAD

**Issue by queequac** [silentmatt/expr-eval#51]
Hi Matt,

I have checked the latest parser... unfortunately the parser cannot parse arbitrary combinations of ternary operators.

Sampe:
`var exp = Parser.parse(' a == 1 ? b == 2 ? 3 : 4 : 5');` -> fails, because when the parser reaches the second question mark, it expected a colon.
`var exp = Parser.parse(' a == 1 ? (b == 2 ? 3 : 4 ) : 5');` -> works, because the parser sees the brackets tying the nested condition together.

Nevertheless, while the first expression is hard to read, it's still correct from a mathematical point of view.

Note: `var exp = Parser.parse(' a == 1 ? 5 :  b == 2 ? 3 : 4');` works again, since it reads the colon after the question mark as "expected".

**Comment by silentmatt** [comment]
Good catch. I was thinking that was a syntax error in JavaScript, but you're right, it should work.
